tools/topology/topology2/platform/intel/dmic-generic.conf

```diff
@@ -211,50 +211,59 @@ IncludeByKey.PASSTHROUGH {
 					type		dai_out
 					stream_name	$DMIC0_NAME
 					node_type $DMIC_LINK_INPUT_CLASS
-					num_input_audio_formats	6
-					Object.Base.input_audio_format [
-						{
-							in_rate			$DMIC0_RATE
-							in_bit_depth		16
-							in_valid_bit_depth	16
-						}
-						{
-							in_rate			$DMIC0_RATE
-							in_bit_depth		32
-							in_valid_bit_depth	24
-							in_sample_type		$SAMPLE_TYPE_MSB_INTEGER
-						}
-						{
-							in_rate			$DMIC0_RATE
-							in_bit_depth		32
-							in_valid_bit_depth	32
-						}
-						{
-							in_rate			$DMIC0_RATE
-							in_channels		4
-							in_bit_depth		16
-							in_valid_bit_depth	16
-							in_ch_cfg		$CHANNEL_CONFIG_3_POINT_1
-							in_ch_map		$CHANNEL_MAP_3_POINT_1
-						}
-						{
-							in_rate			$DMIC0_RATE
-							in_channels		4
-							in_bit_depth		32
-							in_valid_bit_depth	24
-							in_sample_type		$SAMPLE_TYPE_MSB_INTEGER
-							in_ch_cfg		$CHANNEL_CONFIG_3_POINT_1
-							in_ch_map		$CHANNEL_MAP_3_POINT_1
-						}
-						{
-							in_rate			$DMIC0_RATE
-							in_channels		4
-							in_bit_depth		32
-							in_valid_bit_depth	32
-							in_ch_cfg	$CHANNEL_CONFIG_3_POINT_1
-							in_ch_map	$CHANNEL_MAP_3_POINT_1
-						}
-					]
+					IncludeByKey.NUM_DMICS {
+					"2" {
+						num_input_audio_formats	3
+						Object.Base.input_audio_format [
+							{
+								in_rate			$DMIC0_RATE
+								in_bit_depth		16
+								in_valid_bit_depth	16
+							}
+							{
+								in_rate			$DMIC0_RATE
+								in_bit_depth		32
+								in_valid_bit_depth	24
+								in_sample_type		$SAMPLE_TYPE_MSB_INTEGER
+							}
+							{
+								in_rate			$DMIC0_RATE
+								in_bit_depth		32
+								in_valid_bit_depth	32
+							}
+						]
+					}
+					"4" {
+						num_input_audio_formats	3
+						Object.Base.input_audio_format [
+							{
+								in_rate			$DMIC0_RATE
+								in_channels		4
+								in_bit_depth		16
+								in_valid_bit_depth	16
+								in_ch_cfg		$CHANNEL_CONFIG_3_POINT_1
+								in_ch_map		$CHANNEL_MAP_3_POINT_1
+							}
+							{
+								in_rate			$DMIC0_RATE
+								in_channels		4
+								in_bit_depth		32
+								in_valid_bit_depth	24
+								in_sample_type		$SAMPLE_TYPE_MSB_INTEGER
+								in_ch_cfg		$CHANNEL_CONFIG_3_POINT_1
+								in_ch_map		$CHANNEL_MAP_3_POINT_1
+							}
+							{
+								in_rate			$DMIC0_RATE
+								in_channels		4
+								in_bit_depth		32
+								in_valid_bit_depth	32
+								in_ch_cfg	$CHANNEL_CONFIG_3_POINT_1
+								in_ch_map	$CHANNEL_MAP_3_POINT_1
+							}
+						]
+					}
+					}
```

Comment:
@singalsu for the sake of completeness, shouldnt you also limit the number of output formats to fix the channels?

Reply:
Yep, it would be more clear. We would never in this case have copier changing channels count if it would have the capability.

Reply:
It's now updated.